### PR TITLE
Skip ORC and Parquet readers' benchmark cases that are not currently supported

### PR DESCRIPTION
**Pull request by vuule**
Closes: #9961
Skipping the following:
- ORC reader: row selection through stripe selection
- ORC reader: lists column with row selection
- Parquet reader: row selection through row group selection
- Parquet reader: lists column with row selection

### CODE REVIEW

**Comment by vuule** [comment]
CC @GregoryKimball @randerzander - this should fix the ORC reader benchmark failures, and reenable some Parquet reader cases.

File: cpp/benchmarks/io/orc/orc_reader.cpp [hunk 1]
```diff
@@ -88,13 +88,13 @@ void BM_orc_read_varying_options(benchmark::State& state)
   auto const use_np_dtypes = (flags & 2) != 0;
   auto const ts_type       = cudf::data_type{static_cast<cudf::type_id>(state.range(state_idx++))};
 
+  // No nested types here - cannot use skip_rows with such types
```
Comment (devavret) [inline cpp/benchmarks/io/orc/orc_reader.cpp:91]:
```suggestion
  // skip_rows is not supported on nested types
```

**Review by devavret — APPROVED** [review]
very nitpicky review. completely optional

**Review by davidwendt — APPROVED** [review]

**Comment by vuule** [comment]
@gpucibot merge

**Pull request merged**